src/ingredient/ingredient.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { IngredientDto } from './dto';
+
+@Injectable()
+export class IngredientService {
+  constructor(private prismaService: PrismaService) {}
+
+  async createIngredient(dto: IngredientDto) {
+    return await this.prismaService.ingredient.create({ data: { ...dto } });
+  }
+
+  async getIngredient(id: string) {
+    return await this.prismaService.ingredient.findFirst({
+      where: { id },
+    });
+  }
+  async getIngredients(ids?: string[]) {
```

Comment:
this does break single responsibility principle. If you have optional param or boolean flag in your method, it probably breaks single responsibility principle. Of course it is not always the case, I myself use it a lot as well, but if you try to describe this method, you would describe it like this:

getIngredients method gets all ingredients or ingredients by ingredient ids. And the problem is that I used OR to describe it, meaning that it does 2 things.

I think it would be completely fine, if it was method to find ingredients by restaurant id and there was an optional param ids for filtering. 
Then,  you would describe it: this method finds ingredients by restaurant id with ability to filter ingredients by their ids

Reply:
But if you think, you find out that this method can be described as:
gets all ingredients with ability to filter some of them by their ids. And maybe this method is fine. 

What I recommend is that, try to always separate such methods into different methods if possible. But if your api requires you to have that feature, I guess there might not be much choice .